go/oasis-test-runner/oasis/log.go

```diff
@@ -66,3 +67,15 @@ func LogAssertNoMergeDiscrepancyDetected() log.WatcherHandlerFactory {
 func LogAssertPeerExchangeDisabled() log.WatcherHandlerFactory {
 	return LogAssertEvent(tendermint.LogEventPeerExchangeDisabled, "peer exchange not disabled")
 }
+
+// LogAssertUpgradeStartup returns a handler which checks whether a startup migration
+// handler was run based on JSON log output.
+func LogAssertUpgradeStartup() log.WatcherHandlerFactory {
+	return LogAssertEvent(upgrade.LogEventStartupUpgrade, "startup upgrade ran")
```

Comment:
The error message is shown in case of an error so it should be `expected startup upgrade did not run`. Similar below.